modules/layers/src/geojson-layer/geojson-layer.js

```diff
@@ -159,10 +160,11 @@ export default class GeoJsonLayer extends CompositeLayer {
       updateTriggers
     } = this.props;
 
-    const drawPoints = pointFeatures && pointFeatures.length > 0;
-    const drawLines = lineFeatures && lineFeatures.length > 0;
-    const hasPolygonLines = polygonOutlineFeatures && polygonOutlineFeatures.length > 0;
-    const hasPolygon = polygonFeatures && polygonFeatures.length > 0;
+    const drawPoints = (pointFeatures && pointFeatures.length > 0) || subLayerProps;
+    const drawLines = (lineFeatures && lineFeatures.length > 0) || subLayerProps;
+    const hasPolygonLines =
+      (polygonOutlineFeatures && polygonOutlineFeatures.length > 0) || subLayerProps;
+    const hasPolygon = (polygonFeatures && polygonFeatures.length > 0) || subLayerProps;
```

Comment:
in the old logic, we set a flag like "drawPoints" when there is point feature, by adding "|| subLayerProps" how do we guarantee that? because with subLayerProps maybe there is still no point layer

Reply:
subLayerProps can override the parent layer logic. In this case, no `data` is passed in, and we use pre-generated attribute buffers instead.